In [1]:
import pandas
import seaborn
import matplotlib.pyplot as plt

url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'

chipotle_data = pandas.read_csv(url, sep = '\t')

chipotle_data.head(20)

order_id  quantity                              item_name  \
0          1         1           Chips and Fresh Tomato Salsa   
1          1         1                                   Izze   
2          1         1                       Nantucket Nectar   
3          1         1  Chips and Tomatillo-Green Chili Salsa   
4          2         2                           Chicken Bowl   
5          3         1                           Chicken Bowl   
6          3         1                          Side of Chips   
7          4         1                          Steak Burrito   
8          4         1                       Steak Soft Tacos   
9          5         1                          Steak Burrito   
10         5         1                    Chips and Guacamole   
11         6         1                   Chicken Crispy Tacos   
12         6         1                     Chicken Soft Tacos   
13         7         1                           Chicken Bowl   
14         7         1                    Chips and Guacamole   
15         8         1  Chips and Tomatillo-Green Chili Salsa   
16         8         1                        Chicken Burrito   
17         9         1                        Chicken Burrito   
18         9         2                            Canned Soda   
19        10         1                           Chicken Bowl   

                                   choice_description item_price  
0                                                 NaN     $2.39   
1                                        [Clementine]     $3.39   
2                                             [Apple]     $3.39   
3                                                 NaN     $2.39   
4   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5   [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                 NaN     $1.69   
7   [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8   [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9   [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25   
10                                                NaN     $4.45   
11  [Roasted Chili Corn Salsa, [Fajita Vegetables,...     $8.75   
12  [Roasted Chili Corn Salsa, [Rice, Black Beans,...     $8.75   
13  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $11.25   
14                                                NaN     $4.45   
15                                                NaN     $2.39   
16  [Tomatillo-Green Chili Salsa (Medium), [Pinto ...     $8.49   
17  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...     $8.49   
18                                           [Sprite]     $2.18   
19  [Tomatillo Red Chili Salsa, [Fajita Vegetables...     $8.75

In [2]:
chipotle_data.describe()

order_id     quantity
count  4622.000000  4622.000000
mean    927.254868     1.075725
std     528.890796     0.410186
min       1.000000     1.000000
25%     477.250000     1.000000
50%     926.000000     1.000000
75%    1393.000000     1.000000
max    1834.000000    15.000000

Based on the method we used,
1. There were 4622 different rows in the data set.
2. On average, people ordered 1 item.
3. The minimum quantity of items ordered is 1.
4. There were 1834 orders in total.
5. The largest quantity of things ordered for only 1 kind of item was 15.
6. Less than 25 % of people ordered more than 1 item.

In [3]:
# listing all the columns and their data types.
chipotle_data.dtypes

order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

In [4]:
# figure out all the items that can be ordered
# how many times each ordered

group = chipotle_data.groupby('item_name').sum()
group = group.sort_values("quantity", ascending=False)
group
most_ordered_item = group.head(1).quantity

# most_ordered_item = group.loc[0:1, ["item_name"]]
print(most_ordered_item)

item_name
Chicken Bowl    761
Name: quantity, dtype: int64


The most ordered item is Chicken Bowl and it was ordered 761 times.

In [15]:
# total revenue from chipotle
total_revenue = 0

for index, row in chipotle_data.iterrows():
    price = float(row.item_price[1:])
    quantity = row.quantity
    total_revenue += price * quantity

print(f"Total Revenue: $ {total_revenue:.2f}")

Total Revenue: $ 39237.02


In [20]:
# number of orders placed
order_ids = chipotle_data.order_id
print(f"The number of orders placed is {max(order_ids.unique())}.")

The number of orders placed is 1834.


In [23]:
# items > $10
items = set()

for index, row in chipotle_data.iterrows():
    price = float(row.item_price[1:])

    if price > 10:
        item_name = row.item_name
        items.add(item_name)

print("Items that cost more than $10.00:\n", items)

Items that cost more than $10.00:
 {'Steak Burrito', 'Carnitas Burrito', 'Veggie Soft Tacos', 'Side of Chips', 'Carnitas Soft Tacos', 'Barbacoa Bowl', 'Chicken Salad Bowl', 'Steak Bowl', 'Steak Salad Bowl', 'Carnitas Crispy Tacos', 'Veggie Burrito', 'Chips and Fresh Tomato Salsa', 'Chicken Salad', 'Chips and Guacamole', 'Chicken Crispy Tacos', 'Steak Crispy Tacos', 'Barbacoa Burrito', 'Carnitas Bowl', 'Chicken Burrito', 'Steak Soft Tacos', 'Veggie Salad Bowl', 'Chicken Soft Tacos', 'Barbacoa Salad Bowl', 'Carnitas Salad Bowl', 'Barbacoa Crispy Tacos', 'Veggie Bowl', '6 Pack Soft Drink', 'Barbacoa Soft Tacos', 'Bottled Water', 'Chicken Bowl', 'Bowl'}
